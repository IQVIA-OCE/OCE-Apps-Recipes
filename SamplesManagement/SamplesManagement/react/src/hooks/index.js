export { useFetcher } from './useFetcher';
export { useBoolean } from './useBoolean';
export { useHandleData } from './useHandleData';
export { useBanner } from './useBanner';
