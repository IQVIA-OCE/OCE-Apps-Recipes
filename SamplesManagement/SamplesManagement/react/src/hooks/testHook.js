import renderer from 'react-test-renderer';
import React from 'react';

const TestHook = ({ callback }) => {
  callback();
  return null;
};

export const testHook = callback =>
  renderer.create(<TestHook callback={callback} />);

const TestRenderHook = ({ callback }) => {
  return callback();
};

export const testRenderHook = callback =>
  renderer.create(<TestRenderHook callback={callback} />);
