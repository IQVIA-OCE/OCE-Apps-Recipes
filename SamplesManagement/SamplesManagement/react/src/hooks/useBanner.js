import { useEffect, useState } from 'react';

export const useBanner = () => {
  const [banner, setBanner] = useState({
    variant: '',
    message: '',
    visible: false,
    icon: '',
  });

  useEffect(() => {
    const timer = setTimeout(
      () => setBanner(prev => ({ ...prev, visible: false })),
      4000
    );

    return () => clearTimeout(timer);
  }, [banner.visible]);

  return [banner, setBanner];
};
