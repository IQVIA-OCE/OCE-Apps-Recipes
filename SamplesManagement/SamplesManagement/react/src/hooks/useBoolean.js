import { useState } from 'react';

export const useBoolean = initialValue => {
  const [value, setValue] = useState(initialValue);

  const toggle = () => setValue(v => !v);
  const setTrue = () => setValue(true);
  const setFalse = () => setValue(false);

  return [value, { toggle, setTrue, setFalse }];
};
