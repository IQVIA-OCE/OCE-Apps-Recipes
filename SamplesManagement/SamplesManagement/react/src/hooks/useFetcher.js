import { useCallback, useEffect, useState } from 'react';

export const useFetcher = (fetch, normalizer) => {
  const [value, setValue] = useState({
    loading: true,
    error: '',
    data: null,
    metadata: null,
  });

  const handleFetch = async () => {
    setValue(prevState => ({
      ...prevState,
      loading: true,
    }));

    try {
      let [data, metadata] = await fetch();
      if (typeof normalizer === 'function') {
        data = normalizer(data);
      }
      setValue({
        loading: false,
        error: '',
        data,
        metadata,
      });
    } catch (error) {
      setValue({
        loading: false,
        error: error,
        data: null,
        metadata: null,
      });
    }
  };

  useEffect(() => {
    handleFetch();
  }, []);

  return [value, { handleFetch, setValue }];
};
