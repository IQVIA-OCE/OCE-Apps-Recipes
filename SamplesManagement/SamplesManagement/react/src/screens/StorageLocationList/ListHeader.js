import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Colors } from 'apollo-react-native';

const ListHeader = () => {
  return (
    <View style={styles.root}>
      <View style={styles.col}>
        <Text style={styles.title}>ADDRESS</Text>
      </View>
      <View style={styles.col}>
        <Text style={styles.title}>DEFAULT STORAGE LOCATION</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderBottomWidth: 1,
    backgroundColor: Colors.themeGrey[50],
    borderColor: Colors.themeGrey[200],
  },
  col: {
    flex: 1,
  },
  title: {
    fontWeight: 'bold',
  },
});

export default ListHeader;
