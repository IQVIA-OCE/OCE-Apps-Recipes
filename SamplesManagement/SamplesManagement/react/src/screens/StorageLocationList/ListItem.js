import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { RadioButton, IconButton, secondaryRed, themePrimary, themeGrey } from 'apollo-react-native';

const ListItem = ({ item, handleChange, handleDelete, handleEdit }) => {
  const { address, isDefault, id } = item;
  return (
    <View style={styles.root}>
      <View style={styles.col}>
        <Text>{address}</Text>
      </View>
      <View style={[styles.col, styles.actionCol]}>
        <RadioButton
          value={id}
          status={isDefault ? 'checked' : 'unchecked'}
          onPress={() => !isDefault && handleChange(id)}
        />
        <View style={styles.actions}>
          <IconButton
            icon="delete"
            color={secondaryRed[400]}
            onPress={() => handleDelete(id)}
          />
          <IconButton
            icon="pencil"
            color={themePrimary[500]}
            onPress={() => handleEdit(id)}
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderBottomWidth: 1,
    borderColor: themeGrey[200],
    alignItems: 'center',
  },
  col: {
    flex: 1,
  },
  actionCol: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  actions: { flexDirection: 'row' },
});

export default ListItem;
