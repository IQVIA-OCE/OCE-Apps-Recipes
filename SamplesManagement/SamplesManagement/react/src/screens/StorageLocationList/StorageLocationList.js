import React, { useCallback } from 'react';
import { StyleSheet, View, FlatList, Platform } from 'react-native';
import FormHeader from '../../components/FormHeader/FormHeader';
import {
  deleteLocation,
  fetchLocationsList,
  updateDefaultLocation,
} from '../../api/StorageLocation';
import { Banner, Text, useTheme } from 'apollo-react-native';
import { environment } from 'oce-apps-bridges';
import ListHeader from './ListHeader';
import ListItem from './ListItem';
import StorageLocation from '../StorageLocation/StorageLocation';
import { normalizer } from '../../utils/utils';
import { useBanner, useBoolean, useFetcher } from '../../hooks';
import Loader from "../../components/Loader/Loader";
import { useNavigation, useRoute, useFocusEffect } from '@react-navigation/native';
import alert from '../../utils/alert';

const keys = {
  OCE__FullAddress__c: 'address',
  OCE__IsDefaultStorageLocation__c: 'isDefault',
  Id: 'id',
};

const StorageLocationList = () => {
  console.log('plat', Platform.OS);
  const [banner, setBanner] = useBanner();
  const userId = environment.userID();
  const [refreshWidget, setRefreshWidgetActions] = useBoolean(false);

  const theme = useTheme();
  const navigation = useNavigation();
  const route = useRoute();

  const [{ loading, error, data }, { handleFetch, setValue }] = useFetcher(
    async () => await fetchLocationsList(userId),
    normalizer(keys)
  );

  useFocusEffect(
    useCallback(() => {
      if (route.params && route.params.refresh) {
        handleFetch();
        setRefreshWidgetActions.setTrue();
      }
      return () => {
        navigation.setParams({
          refresh: false,
        });
      };
    }, [route.params?.refresh])
  );

  const handleAction = async (id, type) => {
    setValue(prevState => ({ ...prevState, loading: true }));
    try {
      let message = 'Success';
      if (type === 'changeDefault') {
        await updateDefaultLocation(id);
        message = 'Default Storage Location has been successfully set.';
      }
      if (type === 'delete') {
        await deleteLocation(id);
        message = `Samples Management Address was deleted`;
      }
      handleFetch();
      setRefreshWidgetActions.setTrue();
      setBanner({
        variant: 'success',
        message,
        visible: true,
        icon: 'checkbox-marked-circle',
      });
    } catch (e) {
      console.warn(e);
      setValue(prevState => ({ ...prevState, loading: false }));
      setBanner({
        variant: 'error',
        message: e.message,
        visible: true,
        icon: 'alert-circle',
      });
    }
  };

  const handleDelete = async id => {
    alert(
      '',
      'Are you sure you want to delete this Samples Management Address?',
      [
        {
          text: 'Cancel',
          onPress: () => console.log('cancel'),
          style: 'cancel',
        },
        { text: 'Delete', onPress: () => handleAction(id, 'delete') },
      ],
      { cancelable: false }
    );
  };

  if (error) {
    return (
      <View>
        <Text>{error.message ? error.message : error}</Text>
      </View>
    );
  }

  return (
    <View style={[styles.root, { backgroundColor: theme.colors.background }]}>
      {loading && <Loader/>}
      <Banner
        variant={banner.variant}
        icon={banner.icon}
        visible={banner.visible}
      >
        {banner.message}
      </Banner>
      <FormHeader
        icon="map-marker"
        label="Samples Management"
        title="Samples Management Addresses"
        controls={[
          {
            label: 'Back',
            onPress: () => {
              navigation.navigate('Dashboard', {
                refreshLocationWidget: refreshWidget,
              });
            },
          },
          {
            label: 'New',
            color: 'primary',
            mode: 'contained',
            onPress: () => navigation.navigate('StorageLocation'),
          },
        ]}
      />
      <FlatList
        ListHeaderComponent={ListHeader}
        data={data}
        refreshing={loading}
        onRefresh={handleFetch}
        renderItem={({ item }) => (
          <ListItem
            item={item}
            handleChange={id => handleAction(id, 'changeDefault')}
            handleDelete={id => handleDelete(id)}
            handleEdit={id =>
              navigation.navigate('StorageLocation', { locationId: id })
            }
          />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
});

export default StorageLocationList;
