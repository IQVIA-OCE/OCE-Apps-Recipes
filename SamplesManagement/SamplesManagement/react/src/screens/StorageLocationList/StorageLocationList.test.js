import React from 'react';
import StorageLocationList from './StorageLocationList';
import renderer, { act } from 'react-test-renderer';
import { useBanner, useFetcher, useBoolean } from '../../hooks';
import { IconButton, RadioButton, Button } from 'apollo-react-native';
import {
  deleteLocation,
  updateDefaultLocation,
} from '../../api/StorageLocation';
import { environment } from 'oce-apps-bridges';
import { Alert } from 'react-native';

jest.mock('../../hooks/index');
jest.mock('../../api/StorageLocation');
jest.spyOn(Alert, 'alert');

let navigation;
const useBannerValue = [
  { variant: '', message: '', visible: false, icon: '' },
  () => {},
];
describe('StorageLocationList', () => {
  beforeEach(() => {
    environment.userID = jest.fn().mockReturnValue('1');
    navigation = {
      addListener: jest.fn().mockImplementation((_, callback) => callback()),
      getParam: jest
        .fn()
        .mockReturnValueOnce(false)
        .mockReturnValue(true),
      navigate: jest.fn(),
    };
    useBoolean.mockReturnValue([false, { setTrue: jest.fn() }]);
    useFetcher
      .mockReturnValueOnce([
        { loading: false, data: [], error: 'error' },
        {
          handleFetch: jest.fn(),
          setValue: jest.fn(),
        },
      ])
      .mockReturnValueOnce([
        {
          loading: false,
          data: [{ address: 'address 1', isDefault: false, id: '1' }],
        },
        {
          handleFetch: jest.fn(),
          setValue: fn => fn(),
        },
      ]);
    useBanner.mockReturnValue(useBannerValue);
  });
  it('should render error', async () => {
    const tree = renderer.create(
      <StorageLocationList navigation={navigation} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render and trigger actions', async () => {
    const promise = Promise.resolve();
    const tree = renderer.create(
      <StorageLocationList navigation={navigation} />
    );

    // press remove icon
    tree.root.findAllByType(IconButton)[0].props.onPress();
    // press delete Alert button
    Alert.alert.mock.calls[0][2][1].onPress();
    expect(deleteLocation).toHaveBeenCalledTimes(1);
    // press remove icon
    tree.root.findAllByType(IconButton)[0].props.onPress();
    // press cancel Alert button
    Alert.alert.mock.calls[1][2][0].onPress();
    expect(deleteLocation).toHaveBeenCalledTimes(1);

    // press edit should call navigate to StorageLocation screen
    tree.root.findAllByType(IconButton)[1].props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith('StorageLocation', {
      locationId: '1',
    });

    // press Back button
    tree.root.findAllByType(Button)[0].props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith('Dashboard', {
      refreshLocationWidget: false,
    });

    // press New button
    tree.root.findAllByType(Button)[1].props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith('StorageLocation');

    // default location changed successfully
    tree.root.findByType(RadioButton).props.onPress();
    expect(updateDefaultLocation).toHaveBeenCalledTimes(1);
    expect(updateDefaultLocation).toHaveBeenCalledWith('1');

    // default location changed error
    updateDefaultLocation.mockRejectedValue('error');
    tree.root.findByType(RadioButton).props.onPress();
    expect(updateDefaultLocation).toHaveBeenCalledTimes(2);

    await act(() => promise);
    // to match snapshot
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
