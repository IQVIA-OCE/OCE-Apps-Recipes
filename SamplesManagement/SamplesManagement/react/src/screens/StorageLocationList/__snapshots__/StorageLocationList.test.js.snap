// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorageLocationList should render and trigger actions 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
        "zIndex": 99,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "overflow": "hidden",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "height": 0,
          }
        }
      />
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onLayout={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "translateY": -0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 0,
              "marginLeft": 16,
              "marginRight": 56,
              "marginTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "maxWidth": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "margin": 8,
                    },
                    Object {
                      "color": "#000000",
                    },
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "margin": 4,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F6F6F6",
        "borderColor": "#DADADA",
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 12,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "height": 35,
            "marginRight": 10,
            "width": 35,
          },
          Object {
            "backgroundColor": "#AEAEAE",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Ôçé
      </Text>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#262626",
            "fontSize": 12,
            "letterSpacing": 2,
          }
        }
      >
        SAMPLES MANAGEMENT
      </Text>
      <Text
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "bold",
          }
        }
      >
        Samples Management Addresses
      </Text>
    </View>
    <View />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 4,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#d9d9d9",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 0,
            "marginLeft": 7,
            "minWidth": 64,
            "opacity": 1,
            "width": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onLayout={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "textAlign": "center",
                      },
                      undefined,
                      false,
                      Object {
                        "color": "#0768FD",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "marginHorizontal": 16,
                        "marginVertical": 6,
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Back
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "borderRadius": 0,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "#0768FD",
            "borderRadius": 4,
            "borderStyle": "solid",
            "elevation": 2,
            "marginLeft": 7,
            "minWidth": 64,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
            "width": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onLayout={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "textAlign": "center",
                      },
                      undefined,
                      false,
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "marginHorizontal": 16,
                        "marginVertical": 6,
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                New
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "borderRadius": 0,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "address": "address 1",
          "id": "1",
          "isDefault": false,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={
          [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#F6F6F6",
              "borderBottomWidth": 1,
              "borderColor": "#DADADA",
              "flexDirection": "row",
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              ADDRESS
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              DEFAULT STORAGE LOCATION
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#DADADA",
              "flexDirection": "row",
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text>
              address 1
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                status="unchecked"
                style={Object {}}
                value="1"
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "borderRadius": 18,
                          "paddingHorizontal": 6,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View>
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0, 0, 0, 0.26)",
                            "fontSize": 20,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ÔêΩ
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                nativeID="animatedComponent"
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "margin": 6,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 18,
                          "height": 36,
                          "width": 36,
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  <View>
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "#E62626",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ÔáÄ
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                nativeID="animatedComponent"
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "margin": 6,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 18,
                          "height": 36,
                          "width": 36,
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  <View>
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "#0768FD",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Ôè´
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`StorageLocationList should render error 1`] = `
<View>
  <Text>
    error
  </Text>
</View>
`;
